### DISCUSSION TITLE
Stealing focus?

### DISCUSSION THREAD

**Discussion by c33howard** [unpoly/unpoly#288] [Q&A] [answered]
I'm opening a popup overlay and I'd like to focus one button within that overlay for ease of keyboard navigation.  I'm doing this:

```
up.on('up:layer:opened', (event) => {
    if (event.origin === document.querySelector('form.object-search')) {
        event.layer.element.querySelector('.mark-all-correct').focus();
    }
});
```

I can see that the element does briefly get focus, but then the `up-popup` steals it back shortly after.  I think this is when the animation finishes, but I'm not seeing a up event triggered at that time.  Is there someway I can control focus when a popup opens?

Thanks!
Craig

**Comment by adam12** [comment, marked as answer]
Have you tried setting `up-focus=".mark-all-correct"` on the element opening the popup? Here's the docs on [up-focus](https://unpoly.com/focus-option).

**Reply by c33howard** [reply]
Missed that!  Thanks.  Much nicer than listening to events in javascript.  :-)
